core/src/main/java/org/apache/accumulo/core/conf/SiteConfiguration.java

```diff
@@ -315,6 +224,6 @@ public void set(Property property, String value) {
    *          property value
    */
   public void set(String key, String value) {
-    getConfiguration().setProperty(key, value);
+    config.put(key, value);
```

Comment:
This is not thread safe.  One possible strategy would be to have a volatile pointer to an immutable map.  This method could always rebuild the map and then set the pointer.  This is an inefficient implementation for this method.  However, I am assuming this method is only used by test and does not need to be efficient.  This approach allows the normal/non-test use to be very efficient and avoid synchronization, while still keeping things sane for multiple threads in test.

Reply:
If possible, we should remove set methods on SiteConfiguration. SiteConfiguration should always be immutable, since it's based on the fixed contents of a file at startup.

Reply:
Nice, removing the method simplified this class even more.